### PR TITLE
[Feature] Pytest Conversion: Files [OSF-8077]

### PR DESCRIPTION
**Pull request by shadowgamefly**
## Purpose

Conversion of API tests to Pytest
Increase in performance

## Changes

Pytest conversion for all test files in Files folder
Grouping of tests for all test files in Files folder

## Side effects

Not Known

## Ticket

https://openscience.atlassian.net/browse/OSF-8077

### CODE REVIEW

File: api_tests/files/serializers/test_file_serializer.py [hunk 1]
```diff
@@ -1,48 +1,66 @@
...
-        self.file = utils.create_test_file(self.node, self.user)
+    @pytest.fixture()
+    def date_created_tz_aware(self, date_created):
+        return date_created.replace(tzinfo=utc)
 
-        self.date_created = self.file.versions.first().date_created
-        self.date_modified = self.file.versions.last().date_created
-        self.date_created_tz_aware = self.date_created.replace(tzinfo=utc)
-        self.date_modified_tz_aware = self.date_modified.replace(tzinfo=utc)
+    @pytest.fixture()
```
Comment (sloria) [inline api_tests/files/serializers/test_file_serializer.py]: Drive-by comment: no need to make these date strings fixtures if they're only going to be used in one test.

File: api_tests/files/views/test_file_list.py [hunk 2]
```diff
@@ -1,138 +1,160 @@
-# -*- coding: utf-8 -*-
-from nose import tools as nt
-
 from framework.auth.core import Auth
```
Comment (Nesiehr) [inline api_tests/files/views/test_file_list.py]: This should go with local imports and the pytest import should be at the top

File: api_tests/files/views/test_file_list.py [hunk 3]
```diff
@@ -1,138 +1,160 @@
...
+
+@pytest.mark.django_db
+class TestFileFiltering:
+
+    @pytest.fixture()
+    def node(self, user):
+        return ProjectFactory(creator=user)
+
+    @pytest.fixture()
+    def file1(self, user, node):
```
Comment (Nesiehr) [inline api_tests/files/views/test_file_list.py]: Personal / nit, change to be file_one, file_two, etc. (or a more descriptive name if possible)

File: api_tests/files/views/test_file_detail.py [hunk 4]
```diff
@@ -478,104 +496,114 @@ def test_listing(self):
...
+
+    @pytest.fixture()
+    def file1(self, user, node):
+        return api_utils.create_test_file(
+            node, user, filename='file1')
+
+    @pytest.fixture()
+    def payload(self, file1):
+        payload = {
             "data": {
```
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: Replace double quotes (") with single quotes (') wherever possible (basically whenever you don't need to do too much escaping)

File: api_tests/files/views/test_file_list.py (hunk 2 above)
Comment (Nesiehr) [inline api_tests/files/views/test_file_list.py]: This should go with local imports & the pytest import should be at the top of the file

File: api_tests/files/views/test_file_detail.py [hunk 5]
```diff
@@ -27,449 +27,467 @@ def _dt_to_iso8601(value):
...
+        assert res.status_code == 401
 
-    def test_visited_file_has_guid(self):
-        guid = self.file.get_guid(create=True)
-        res = self.app.get(self.file_url, auth=self.user.auth)
-        assert_equal(res.status_code, 200)
-        assert_is_not_none(guid)
-        assert_equal(res.json['data']['attributes']['guid'], guid._id)
+        # test_must_be_contributor(self, app, file_url):
+        user = AuthUserFactory()
```
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: Could be renamed to be more specific, non_contrib / non_contributor, etc.

File: api_tests/files/views/test_file_detail.py [hunk 6]
```diff
@@ -27,449 +27,467 @@ def _dt_to_iso8601(value):
...
+        file.save()
+        node.reload()
+        assert res.status_code == 200
+        assert file.checkout == user
+
+        res = app.get(
+            file_url,
+            auth=user.auth
         )
+        assert node.logs.count(),2
```
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: These shouldn't use commas, use ==

File: api_tests/files/views/test_file_detail.py [hunk 7]
```diff
@@ -27,449 +27,467 @@ def _dt_to_iso8601(value):
...
-        self.file.reload()
-        assert_equal(res.status_code, 200)
-        assert_equal(self.file.checkout, None)
+        file.reload()
+        assert res.status_code == 200
+        assert file.checkout == None
 
-    def test_removed_contrib_files_checked_in(self):
+    def test_removed_contrib_files_checked_in(self, app, node, file):
         user = AuthUserFactory()
```
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: Nit, this can be renamed to be read_write_contrib or something more descriptive

File: api_tests/files/views/test_file_detail.py [hunk 8]
```diff
@@ -478,104 +496,114 @@ def test_listing(self):
...
+        ).status_code == 405
+
+@pytest.mark.django_db
+class TestFileTagging:
+    @pytest.fixture()
+    def node(self, user):
+        return ProjectFactory(creator=user)
+
+    @pytest.fixture()
+    def file1(self, user, node):
```
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: Personal / Nit, rename to file_one for consistent naming conventions

File: api_tests/files/views/test_file_list.py (hunk 2 above)
Comment (Nesiehr) [inline api_tests/files/views/test_file_list.py]: This should go with local imports and the pytest import should be at the top

File: api_tests/files/views/test_file_list.py (hunk 3 above)
Comment (Nesiehr) [inline api_tests/files/views/test_file_list.py]: Personal / Nit, change to be file_one, file_two, etc. (or a more descriptive name if possible)

File: api_tests/files/views/test_file_detail.py (hunk 5 above)
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: Could be renamed to be more specific, non_contrib / non_contributor, etc.

File: api_tests/files/views/test_file_detail.py (hunk 6 above)
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: These shouldn't use commas, use ==

File: api_tests/files/views/test_file_detail.py (hunk 7 above)
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: Nit, this can be renamed to be read_write_contrib or something more descriptive

File: api_tests/files/views/test_file_detail.py (hunk 8 above)
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: Personal / Nit, rename to file_one for consistent naming conventions

File: api_tests/files/views/test_file_detail.py (hunk 4 above)
Comment (Nesiehr) [inline api_tests/files/views/test_file_detail.py]: Replace double quotes (") with single quotes (') wherever possible (basically whenever you don't need to do too much escaping)

Comment (alexschiller) [inline api_tests/files/views/test_file_detail.py]: @shadowgamefly for your reference: http://cosdev.readthedocs.io/en/latest/style_guides/python.html#general-naming-guidelines

**Comment by alexschiller** [comment]
@shadowgamefly Did you mean to close this PR?

**Comment by shadowgamefly** [comment]
@alexschiller I accidentally closed this yesterday, so I just opened another one....

**Comment by shadowgamefly** [comment]
reference to pull request #7291
